packages/messages/src/browser/notifications.ts

```diff
@@ -72,8 +74,16 @@ export class Notifications {
                 button.addEventListener('click', () => {
                     action.fn(handler);
                     close();
+                    clearTimeout(this.closeTimer);
                 });
             }
+            if (properties.timeout !== undefined && properties.timeout > 0) {
```

Comment:
you can move this `if` statement up before L70 to make a local `closeTimer`. 
something like  `const closeTimer = !!properties.timeout ? window.setTimeout() : undefined` should work.
then you can use it in click handler.

Reply:
Move the closeTimer before and use it as a const